src/main/scala/resource/package.scala

```diff
@@ -77,6 +77,56 @@ package object resource {
     toReturn
   }
 
+  /**
+    * Creates a [[ManagedResource]] shared by all users for any type with a [[Resource]] type class implementation.
+    *
+    * There is only one instance of the resource at the same time for all the users.
+    * The instance will be closed once no user is still using it.
```

Comment:
We should add some concurrency notes, e.g. if the open call is slightly after a close, you could end up closing/reopening the resource.

Also, the resource should have a thread safe api because we don't serialize access to it, I.e. Multiple threads can touch the resource